core/amazon/index.js

```diff
@@ -165,6 +174,11 @@ var amazonProduct = {
                                             && item.ItemAttributes[0].Title && item.ItemAttributes[0].Title[0]
                                         }
                                     } else {
+                                        // Check if variation values are too long or too numerous for selection through conversation
+                                        if (value.length > 20 || Object.keys(ref).length == 9) {
```

Comment:
Same as above, this will allow lengths of 9 because I'm checking the length before adding the value.... but I'll move the check after adding the value so that's it's not confusing... just thought I'd avoid constantly resetting conversational to true after crossing the threshold